packages/wpcom-checkout/src/checkout-line-items.tsx

```diff
@@ -226,14 +226,61 @@ function WPNonProductLineItem( {
 	/* eslint-enable wpcalypso/jsx-classname-namespace */
 }
 
-function GSuiteUsersList( { product }: { product: ResponseCartProduct } ) {
-	const users = product.extra?.google_apps_users ?? [];
+function EmailMeta( { product, isRenewal }: { product: ResponseCartProduct; isRenewal: boolean } ) {
+	const translate = useTranslate();
+
+	if ( isRenewal ) {
+		let numberOfMailboxes = null;
+
+		if ( isGSuiteOrExtraLicenseProductSlug( product.product_slug ) ) {
+			numberOfMailboxes = product.volume ?? null;
+		}
+
+		if ( isGoogleWorkspaceProductSlug( product.product_slug ) || isTitanMail( product ) ) {
+			numberOfMailboxes = product.current_quantity ?? null;
+		}
+
+		if ( numberOfMailboxes === null ) {
+			return;
+		}
+
+		return (
+			<LineItemMeta>
+				{ translate(
+					'%(numberOfMailboxes)d mailbox for %(domainName)s',
+					'%(numberOfMailboxes)d mailboxes for %(domainName)s',
+					{
+						args: {
+							numberOfMailboxes,
+							domainName: product.meta,
+						},
+						count: numberOfMailboxes,
+					}
+				) }
+			</LineItemMeta>
+		);
+	}
+
+	let mailboxes = [];
+
+	if (
+		isGoogleWorkspaceProductSlug( product.product_slug ) ||
+		isGSuiteOrExtraLicenseProductSlug( product.product_slug )
+	) {
+		mailboxes = product.extra?.google_apps_users ?? [];
+	}
+
+	if ( isTitanMail( product ) ) {
+		mailboxes = product.extra?.email_users ?? [];
+	}
+
 	return (
 		<>
-			{ users.map( ( user, index ) => {
+			{ mailboxes.map( ( mailbox, index ) => {
 				return (
-					<LineItemMeta key={ user.email }>
-						<div key={ user.email }>{ user.email }</div>
+					<LineItemMeta key={ mailbox.email }>
+						<div key={ mailbox.email }>{ mailbox.email }</div>
+
 						{ index === 0 && <GSuiteDiscountCallout product={ product } /> }
```

Comment:
Side note: `GSuiteDiscountCallout` checks for the product internally, so is safe to use for any product type.